src/js/components/projects/createForm.tsx

```diff
@@ -69,15 +89,153 @@ const ProjectForm = ({
+      setRepoBranches(baseBranches);
+      setFetchingBranches(false);
+    }
+  };
+
+  const setBranch = (branch: string) => {
+    setInputs({ ...inputs, branch_name: branch });
+  };
+
+  const resetBranchForm = () => {
+    setBranch('');
+    resetFilterVal();
+  };
+
+  const closeBranchForm = () => {
+    resetBranchForm();
+    setFromBranchChecked(false);
+  };
+
+  const handleBranchCheckboxChange = (
+    event: any,
+    { checked }: { checked: boolean },
+  ) => {
+    if (checked) {
+      doGetBranches();
+      setFromBranchChecked(true);
+    } else {
+      closeBranchForm();
+    }
+  };
+
+  const handleBranchSelection = (
+    event: any,
+    { selection }: { selection: ComboboxOption[] },
+  ) => {
+    setBranch(selection[0].label);
+  };
+
+  const handleBranchChange = (event: any, { value }: { value: string }) => {
+    setFilterVal(value);
+  };
+
+  const handleBranchRemoveSelection = () => {
+    resetBranchForm();
+  };
+
+  const noFeatureBranches = !repoBranches.length;
+  const inputVal = inputs.branch_name;
+  const branchOptions: ComboboxOption[] = repoBranches.map((item, index) => ({
+    id: `${index + 1}`,
+    label: item,
+  }));
+  const selection = branchOptions.find((branch) => branch.label === inputVal);
+  let noOptionsFoundText = null;
+  if (fetchingBranches) {
+    noOptionsFoundText = i18n.t('Loading existing branches…');
+  } else {
+    noOptionsFoundText = (
+      <>
+        {noFeatureBranches
+          ? i18n.t("There aren't any available branches at this time.")
+          : i18n.t('No matching branches found.')}{' '}
+        <Button
+          label={i18n.t('Check again')}
+          variant="link"
+          onClick={doGetBranches}
+        />{' '}
+        {i18n.t('to refresh this list or')}{' '}
+        <Button
+          label={i18n.t('start from a new branch.')}
+          variant="link"
+          onClick={closeBranchForm}
+        />
+      </>
+    );
+  }
+
   return (
     <form onSubmit={handleSubmit} className="slds-form slds-m-bottom--large">
       {isOpen && (
         <>
+          <p className="slds-form-element__help slds-m-bottom_small">
+            <Trans i18nKey="createProjectHelpText">
+              Projects in Metecho are the high-level features that can be broken
+              down into smaller parts when creating Tasks. You can create a new
+              project or create a project based on an existing GitHub branch.
+              Every project requires a unique project name, which becomes the
+              branch name in GitHub unless you choose to use an existing branch.
+            </Trans>
+          </p>
+          <Checkbox
+            id="project-base-branch"
+            labels={{
+              label: `${i18n.t('Use existing GitHub branch')}`,
+            }}
+            checked={fromBranchChecked}
+            className="slds-form-element_stacked slds-p-left_none"
+            onChange={handleBranchCheckboxChange}
+          />
```

Comment:
@stacyk In our mtg today, David and Jason suggested making this a set of two radio options instead of a checkbox, so that it's clear what the two options are (create from existing, or create a new branch). It's a pretty simple change for me to make, unless you have concerns about it?